x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/nio/SSLOutboundBuffer.java

```diff
@@ -74,7 +68,24 @@ boolean hasEncryptedBytesToFlush() {
 
     @Override
     public void close() {
-        IOUtils.closeWhileHandlingException(currentPage);
-        IOUtils.closeWhileHandlingException(pages);
+        Exception closeException = null;
+        try {
+            IOUtils.close(currentPage);
+        } catch (Exception e) {
+            closeException = e;
+        }
+        currentPage = null;
+        Page p;
+        while ((p = pages.pollFirst()) != null) {
+            try {
+                p.close();
+            } catch (Exception ex) {
+                closeException = ExceptionsHelper.useOrSuppress(closeException, ex);
```

Comment:
Might we be closing a lot of pages here? If so, should we bound the number of suppressed exceptions here, just in case?

Alternatively, do we need to track all the exceptions and/or to carry on closing all the things after the first exception? We could just `assert false` in each catch block if not.

Reply:
> Might we be closing a lot of pages here?

I don't think we can really have all that many pages here since this is just the SSL outbound buffer which IMO would have a bug if it were to grow boundlessly (and it seems we are taking proper precautions to not continue to encrypt stuff if writing is backed up on a channel).